mlflow/sagemaker/__init__.py

```diff
@@ -135,35 +150,78 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
     os.system(cmd)
 
 
-def deploy(app_name, model_path, execution_role_arn, bucket, run_id=None,
-           image="mlflow_sage", region_name="us-west-2"):
+def deploy(app_name, model_path, execution_role_arn=None, bucket=None, run_id=None,
+           image_url=None, region_name="us-west-2"):
     """
-    Deploy model on Sagemaker.
+    Deploy model on SageMaker.
     Current active AWS account needs to have correct permissions setup.
 
-    :param app_name: Name of the deployed app.
+    :param app_name: Name of the deployed application.
     :param path: Path to the model.
-                Either local if no run_id or MLflow-relative if run_id is specified)
-    :param execution_role_arn: Amazon execution role with sagemaker rights
-    :param bucket: S3 bucket where model artifacts are gonna be stored
+                 Either local if no run_id or MLflow-relative if run_id is specified)
+    :param execution_role_arn: Amazon execution role with sagemaker rights. defaults
+                               to the currently-assumed role.
+    :param bucket: S3 bucket where model artifacts will be stored. defaults to a
+                   SageMaker-compatible bucket name.
     :param run_id: MLflow run id.
-    :param image: name of the Docker image to be used.
-    :param region_name: Name of the AWS region to deploy to.
+    :param image: name of the Docker image to be used. if not specified, uses a 
+                  publicly-available pre-built image.
+    :param region_name: Name of the AWS region to which to deploy the application.
     """
+    if not image_url:
+        image_url = _get_prebuilt_image_url(region_name)
+
+    if not execution_role_arn:
+        execution_role_arn = _get_assumed_role_arn()
+
+    if not bucket:
+        eprint("No model data bucket specified, using the default bucket") 
+        bucket = _get_default_s3_bucket(region_name)
+
     prefix = model_path
     if run_id:
         model_path = _get_model_log_dir(model_path, run_id)
-        prefix = run_id + "/" + prefix
+        prefix = os.path.join(run_id, prefix)
     run_id = _check_compatible(model_path)
+
     model_s3_path = _upload_s3(local_model_path=model_path, bucket=bucket, prefix=prefix)
     _deploy(role=execution_role_arn,
-            image=image,
+            image_url=image_url,
             app_name=app_name,
             model_s3_path=model_s3_path,
             run_id=run_id,
             region_name=region_name)
 
 
+def delete(app_name, region_name="us-west-2"):
+    """
+    :param app_name: Name of the deployed application.
```

Comment:
Let's document here that it also deletes all SageMaker Models & EndpointConfigs associated with the Endpoint.